### PR TITLE
[feature/cumulative--10.0.0] AB#33500 - Closed Checkbox & Switch Gap

### PR DESCRIPTION
**Pull request by cbrewer1689**

### CODE REVIEW

**Comment by morethanfire** [comment]
Please do not merge, but please go ahead and review. The PR is contingent on PR #233 being merged first and then this base branch change to `feature/cumulative--10.0.0`.

**Comment by cbrewer1689** [comment]
The label, do not merge, has been removed because #233 has been merged.

File: src/inputs/switch/switch.jsx [hunk 1]
```diff
@@ -0,0 +1,458 @@
...
+    BEM_SWITCH,
+    ENTER_KEY_CODE,
+    UI_CLASS_NAME,
+} from '../../global/constants';
+import Typography from '../../dataDisplay/typography';
+import withStyles from '../../styles/withStyles';
+
+const propTypes = {
+    /**
+     * If `true`, the checkbox will be checked.
```
Comment (groberts314) [inline src/inputs/switch/switch.jsx:18]: "checkbox" => "Switch", right?

File: src/inputs/switch/switch.jsx [hunk 2]
```diff
@@ -0,0 +1,458 @@
...
+        } = this.state;
+
+        if (!disable) {
+            if (isFunction(onChange)) {
+                onChange(id, !isChecked, event);
+            } else {
+                this.setState({
+                    isChecked: !isChecked,
+                }, () => {
+                    // eslint-disable-next-line no-underscore-dangle
```
Comment (groberts314) [inline src/inputs/switch/switch.jsx:337]: Looks like we can remove this now, right?

File: src/versions.js [hunk 3]
```diff
@@ -16,13 +31,6 @@ const versions = {
                     designLibraryDoc: 'https://www.sketch.com/s/8016dbb8-1bb5-4fac-bea4-09f39aca414a',
                 },
```
Comment (groberts314) [inline src/versions.js:32]: Should we add `switch` as well?

**Review by groberts314 — CHANGES_REQUESTED** [review]
Looks pretty good.  A few minor things here and there.

File: docs/src/inputs/switch/switch.jsx [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+import { default as rootDoc } from '!!@advclb/react-docgen-loader!react-cm-ui/inputs/switch/switch';
+/* eslint-enable import/no-named-default, import/extensions */
+
+const propTypes = {
+    location: PropTypes.shape({
+        pathname: PropTypes.string,
+    }).isRequired,
+};
+
+function DocsCheckbox(props) {
```
[the remark this answers is not in the document]
Comment (cbrewer1689) [reply docs/src/inputs/switch/switch.jsx]: Fixed.

File: src/inputs/checkbox/checkbox.jsx [hunk 5]
```diff
@@ -1,69 +1,285 @@
...
+    BEM_CHECKBOX,
+    ENTER_KEY_CODE,
+    UI_CLASS_NAME,
+} from '../../global/constants';
 import Icon from '../../dataDisplay/icon';
+import withStyles from '../../styles/withStyles';
 
 const propTypes = {
+    /**
+     * Whether to align the checkbox to the right or left of a container.
```
[the remark this answers is not in the document]
Comment (cbrewer1689) [reply src/inputs/checkbox/checkbox.jsx]: Fixed.

File: src/inputs/checkbox/checkbox.jsx [hunk 6]
```diff
@@ -1,69 +1,285 @@
...
+     */
     align: PropTypes.oneOf(['left', 'right']),
+    /**
+     * If `true`, the checkbox will be checked.
```
[the remark this answers is not in the document]
Comment (cbrewer1689) [reply src/inputs/checkbox/checkbox.jsx]: Fixed.

File: src/inputs/switch/switch.jsx (hunk 1 above)
Comment (cbrewer1689) [reply src/inputs/switch/switch.jsx]: Fixed.

File: src/inputs/switch/switch.jsx (hunk 2 above)
Comment (cbrewer1689) [reply src/inputs/switch/switch.jsx]: Fixed.

File: src/versions.js (hunk 3 above)
Comment (cbrewer1689) [reply src/versions.js:138]: Fixed.

Comment (cbrewer1689) [reply src/versions.js:138]: @groberts314, I change this file pretty simnifically. It's my belief now, that it provides a little more information a little more accurately.

**Comment by JLeeC** [comment]
PR Review Checkpoint Meeting Thurs 2020-11-10

@groberts314   to re-review. Target: 11/12/2020

File: docs/src/app/app.jsx [hunk 7]
```diff
@@ -59,6 +60,8 @@ const styles = (theme) => ({
     },
 });
 
+console.log(moment.tz.guess());
```
Comment (groberts314) [inline docs/src/app/app.jsx:63]: Uh oh!  We have a rogue `console.log()`.  Let's remove that please!  And likely the import as well if it's not needed.

File: docs/src/inputs/switch/switch.jsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (groberts314) [reply docs/src/inputs/switch/switch.jsx]: Much better, thank you sir!  Resolving...

File: src/inputs/checkbox/checkbox.jsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (groberts314) [reply src/inputs/checkbox/checkbox.jsx]: Looks good.  Resolving.

File: src/inputs/checkbox/checkbox.jsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (groberts314) [reply src/inputs/checkbox/checkbox.jsx]: Looks good.  Resolving.

File: src/inputs/switch/switch.jsx (hunk 1 above)
Comment (groberts314) [reply src/inputs/switch/switch.jsx]: Looks good.  Resolving.

File: src/inputs/switch/switch.jsx (hunk 2 above)
Comment (groberts314) [reply src/inputs/switch/switch.jsx]: Looks good.  Resolving.

File: src/versions.js [hunk 8]
```diff
@@ -1,45 +1,220 @@
...
+                        {
+                            doc: 'https://www.sketch.com/s/6c8808fa-bf58-42e5-a0be-7e9d5798fc5f/a/2ooq5J',
+                            name: 'Button & Single Line Input',
+                            version: '2.0.0',
+                        },
+                    ],
+                    devVersion: '2.0.0',
+                },
+                prompt: {
+                    designArea: 'Button Dropdowns',
```
Comment (groberts314) [inline src/versions.js:72]: Is this `designArea` correct?

File: src/versions.js (hunk 3 above)
Comment (groberts314) [reply src/versions.js:145]: Sounds good @cbrewer1689 .  I see switch is in here now, so I'm gonna resolve this thread.

**Comment by groberts314** [comment]
@cbrewer1689 @JLeeC I have re-reviewed.  I have one outstanding change request (remove a `console.log()` and a related import, that were probably used for debugging and introduced here by accident) and one question about the updated `versions.js`.  Otherwise, it is looking pretty good.

File: docs/src/app/app.jsx (hunk 7 above)
Comment (cbrewer1689) [reply docs/src/app/app.jsx:63]: Oppsi-whoppsi

File: src/versions.js (hunk 8 above)
Comment (cbrewer1689) [reply src/versions.js:72]: Yep, it matches designs docs. Maybe eventually they'll move it, but I probably have to prod them a little bit.

File: docs/src/app/app.jsx (hunk 7 above)
Comment (cbrewer1689) [reply docs/src/app/app.jsx]: Fixed.

Comment (groberts314) [reply docs/src/app/app.jsx]: Looks good.  Resolving.

**Review by groberts314 — APPROVED** [review]

**Pull request merged**